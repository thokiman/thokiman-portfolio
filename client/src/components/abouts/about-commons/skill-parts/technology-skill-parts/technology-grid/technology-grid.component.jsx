import React from 'react';
import {
  TechnologyGridContainer,
  TechnologyGridContent,
  TechnologyImage,
  TechnologyTitle,
} from './technology-grid.styles';

export const TechnologyGrid = ({ technologySkill: { name, itemsPoint } }) => {
  return (
    <TechnologyGridContainer>
      <TechnologyTitle>{name}</TechnologyTitle>
      <TechnologyGridContent>
        {itemsPoint.map(({ id, field, iconPath }) => {
          return <TechnologyImage key={id} src={iconPath} alt={field} />;
        })}
      </TechnologyGridContent>
    </TechnologyGridContainer>
  );
};

export default TechnologyGrid;
