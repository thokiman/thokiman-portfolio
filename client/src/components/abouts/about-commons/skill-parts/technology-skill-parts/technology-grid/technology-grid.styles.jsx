import styled, { keyframes } from 'styled-components';

const colorForegroundNeon = '#60626E'
const colorBackgroundNeon = '#7EC7D4'

const neon = keyframes`
 from {
    box-shadow: 0 0 10px ${ colorForegroundNeon }, 0 0 20px ${ colorForegroundNeon }, 0 0 30px ${ colorForegroundNeon }, 0 0 40px ${ colorBackgroundNeon },
      0 0 70px ${ colorBackgroundNeon }, 0 0 80px ${ colorBackgroundNeon }, 0 0 100px ${ colorBackgroundNeon }, 0 0 150px ${ colorBackgroundNeon };
  }
  to {
    box-shadow: 0 0 5px ${ colorForegroundNeon }, 0 0 10px ${ colorForegroundNeon }, 0 0 15px ${ colorForegroundNeon }, 0 0 20px ${ colorBackgroundNeon },
      0 0 35px ${ colorBackgroundNeon }, 0 0 40px ${ colorBackgroundNeon }, 0 0 50px ${ colorBackgroundNeon }, 0 0 75px ${ colorBackgroundNeon };
  }`

export const TechnologyGridContainer = styled.div`
  position: relative;
  top: 10px;
  margin: 20px auto;
  padding: 0 30px;
  width: 100%;
  height: 100%;
  display: grid;
  grid-template-columns: 1fr;
  grid-template-rows: 1fr;
  gap: 40px 0px;
  justify-content: start;
  justify-items: center;
  grid-template-areas:
    'TechnologyTitle'
    'TechnologyGridContent';
  align-items: flex-start;

  @media only screen and (max-width: 800px) {
    top: 0;
  }
`;

export const TechnologyTitle = styled.div`
  text-align: left;
  font-family: baskerville-old-face;
  font-size: 3vw;
  margin: auto 0;
  font-kerning: auto;
  letter-spacing: 0.01em;
  color: #e3e1e4;
  justify-self: start;

  grid-area: TechnologyTitle;
  @media only screen and (max-width: 800px) {
    font-size: 25px;
  }
`;

export const TechnologyGridContent = styled.div`
  grid-area: TechnologyGridContent;
  margin-bottom:20px;
  
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  grid-auto-rows: 1fr;
  gap: 50px 50px;
  justify-items: center;
`;

export const TechnologyImage = styled.img`
  padding: 0.5vw;
  width: 10vw;
  height: 17vh;
  justify-self: stretch;
  -webkit-animation: ${ neon } 1.5s ease-in-out infinite alternate;
  -moz-animation: ${ neon } 1.5s ease-in-out infinite alternate;
  animation: ${ neon } 1.5s ease-in-out infinite alternate;
  @media only screen and (max-width: 800px) {
    width: 100px;
    height: 100px;
  }
`;
TechnologyGridContainer.displayName = 'TechnologyGridContainer';
TechnologyTitle.displayName = 'TechnologyTitle';
TechnologyGridContent.displayName = 'TechnologyGridContent';

TechnologyImage.displayName = 'TechnologyImage';
