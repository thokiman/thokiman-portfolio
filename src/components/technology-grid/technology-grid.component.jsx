import React from "react";

import "./technology-grid.styles.scss";

const TechnologyGrid = ({ technologySkill: { name, itemsPoint } }) => {
  return (
    <div className="technology-grid-container">
      <div className="technology-title">{name}</div>
      <div className="technology-grid">
        {itemsPoint.map(({ id, field, iconPath }) => {
          return (
            <div key={id} className="technology-grid-item">
              <img src={iconPath} alt={field} className="technology-image" />
            </div>
          );
        })}
      </div>
    </div>
  );
};

export default TechnologyGrid;
