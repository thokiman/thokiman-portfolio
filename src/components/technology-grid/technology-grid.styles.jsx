import styled from "styled-components";

export const TechnologyGridContainer = styled.div`
  position: relative;
  top: 10px;
  margin: 20px auto;
  padding: 0 30px;
  width: 100%;
  height: 100%;
  display: flex;
  flex-direction: row;
  align-items: center;
  @media only screen and (max-width: 800px) {
    top: 0;
  }
`;

export const TechnologyTitle = styled.div`
  text-align: left;
  font-family: baskerville-old-face;
  font-size: 3vw;
  margin: auto 0;
  font-kerning: auto;
  letter-spacing: 0.01em;
  color: #e3e1e4;
  width: 15%;
  @media only screen and (max-width: 800px) {
    font-size: 25px;
  }
`;

export const TechnologyGridContent = styled.div`
  position: relative;
  left: 140px;
  bottom: 10px;
  display: grid;
  gap: 10px 20px;
  grid-template-columns: repeat(3, 1fr);
  grid-auto-rows: 1fr;
  justify-content: center;
  align-items: center;
  width: 30vw;
  @media only screen and (max-width: 800px) {
    justify-content: flex-end;
    left: 175px;
  }
`;

export const TechnologyGridItem = styled.div`
  display: flex;
  justify-content: center;
  width: 100%;
  height: 100%;
`;

export const TechnologyImage = styled.img`
  width: 80%;
  height: 100%;
  @media only screen and (max-width: 800px) {
    width: 55px;
    height: 55px;
  }
`;
TechnologyGridContainer.displayName = "TechnologyGridContainer";
TechnologyTitle.displayName = "TechnologyTitle";
TechnologyGridContent.displayName = "TechnologyGridContent";
TechnologyGridItem.displayName = "TechnologyGridItem";
TechnologyImage.displayName = "TechnologyImage";
