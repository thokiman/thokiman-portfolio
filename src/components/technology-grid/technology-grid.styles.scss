.technology-grid-container {
    position: relative;
    top: 10px;
    margin: 20px auto;
    padding: 0 30px;
    width: 100%;
    height: 100%;
    display: flex;
    flex-direction: row;
    align-items: center;





    .technology-title {

        margin: auto 5px;
        text-align: left;
        position: relative;
        top: -2px;
        left: -11px;
        font-family: baskerville-old-face;
        font-size: 300%;
        font-kerning: auto;
        letter-spacing: 0.01em;
        color: #e3e1e4;
        width: 15%;
    }

    .technology-grid {

        position: relative;
        left: 140px;
        bottom: 10px;
        display: grid;
        gap: 10px 10px;
        grid-template-columns: 1fr 1fr 1fr;
        grid-auto-rows: 1fr;
        justify-content: center;
        align-items: center;
        width: 40%;


        .technology-grid-item {

            display: flex;
            justify-content: center;
            width: 100%;
            height: 100%;

            .technology-image {

                width: 80%;
                height: 100%;

            }
        }
    }


}