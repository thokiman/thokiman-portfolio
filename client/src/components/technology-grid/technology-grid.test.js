import React from "react";
import { shallow } from "enzyme";

import { TechnologyGrid } from "./technology-grid.component";

describe("TechnologyGrid component", () => {
  let wrapper;
  const mockText = "";
  const mockArray = [];

  beforeEach(() => {
    const mockProps = {
      technologySkill: { name: mockText, itemsPoint: mockArray },
    };
    wrapper = shallow(<TechnologyGrid {...mockProps} />);
  });

  it("should render TechnologyGrid component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
