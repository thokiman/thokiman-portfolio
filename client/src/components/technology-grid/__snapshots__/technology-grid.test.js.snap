// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TechnologyGrid component should render TechnologyGrid component 1`] = `ShallowWrapper {}`;
